common/config/config.cpp

```diff
@@ -0,0 +1,114 @@
+#include "config.hpp"
+#include "sdmc/sdmc.hpp"
+#include "minIni/minIni.h"
+#include <cstdio>
+
+namespace config {
+
+namespace {
+
+const char CONFIG_PATH[]{"/config/sys-tune/config.ini"};
+// blacklist uses it's own config file because eventually a database
+// may be setup and users can easily update their blacklist by downloading
+// an updated blacklist.ini.
+// Also, the blacklist lookup needs to be as fast as possible
+// (literally a race until the title opens audren), so a seperate, smaller file is ideal.
+const char BLACKLIST_PATH[]{"/config/sys-tune/blacklist.ini"};
+
+void create_config_dir() {
+    /* Creating directory on every set call looks sus, but the user may delete the dir */
+    /* whilst the sys-mod is running and then any changes made via the overlay */
+    /* is lost, which sucks. */
+    sdmc::CreateFolder("/config");
+    sdmc::CreateFolder("/config/sys-tune");
+}
+
+auto get_tid_str(u64 tid) -> const char* {
+    static char buf[21]{};
+    std::sprintf(buf, "%016lX", tid);
+    return buf;
+}
+
+} // namespace
```

Comment:
Not a fan of those. Does your formatter do that?

Reply:
No, that's me. I'll remove them